hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecord.java

```diff
@@ -64,7 +68,8 @@ public static <T> BufferedRecord<T> forRecordWithContext(HoodieRecord<T> record,
     } catch (IOException e) {
       throw new HoodieException("Failed to get isDelete from record.", e);
     }
-    return new BufferedRecord<>(recordKey, record.getOrderingValue(schema, props), record.getData(), schemaId, isDelete);
+    T row = record.getData();
+    return new BufferedRecord<>(recordKey, readerContext.getOrderingValue(row, schema, orderingFieldName), row, schemaId, isDelete);
```

Comment:
we should use `readerContext.getOrderingValue(` instead of `record.getOrderingValue(` to unify the ordering value format in `BufferedRecord`, `record.getOrderingValue(` is engine agnostic.